hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -168,4 +173,28 @@ public String retryQuestion(String questionId, String comments, Question questio
                 .build());
         return askQuestion(question);
     }
+
+    public BinaryResource getAvatar(String agentId)
+    {
+        try
+        {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(format(config.getAvatarUrl(), agentId)))
+                    .headers(authService.getAuthHeaders())
+                    .GET()
+                    .build();
+
+            HttpResponse<InputStream> httpResponse = client.send(request, BodyHandlers.ofInputStream());
+
+            ensureCorrectHttpStatusReturned(SC_OK, httpResponse);
+            log.atDebug().log("Agent with id {} received avatar successfully", agentId);
+
+            File tempImageFile = TempFileProvider.createTempFile(httpResponse.body(), format("avatar-%s", agentId), "png");
+            return new FileBinaryResource(tempImageFile);
+        }
+        catch (Exception e)
+        {
+            throw new WebScriptException(SC_SERVICE_UNAVAILABLE, format("Failed to get avatar for agent with id %s", agentId), e);
```

Comment:
When we throw this exception, how does the response look like to the client that requested an avatar with e.g.: a bad agent Id?

Reply:
**Note:** should be verified via E2E test if possible IMO, as requested here: https://github.com/Alfresco/hxinsight-connector/pull/526#discussion_r1694777577

Reply:
Added e2e test to cover it.